### DISCUSSION TITLE
how i've just built customtkinter to 1 exe file

### DISCUSSION THREAD

**Discussion by ForestWildRabbit** [TomSchimansky/CustomTkinter#940] [Ideas]
pyinstaller --noconfirm --onedir --windowed --add-data "<CustomTkinter Location>/customtkinter;customtkinter/" --paths "path\to\venv\Lib\site-packages" main.py --onefile

Script run from virtual enviroment. Operation system is Windows 10.

**Comment by poneoneo** [comment]
i've you tried auto-py-to-exe module instead of pyinstaller ? its more easy to for get an exe file from python code this module working with a gui therefore is simple to configure your exe file and his environements  or packages. Here link for download module  _[](https://pypi.org/project/auto-py-to-exe/)_

or could directly typed : _pip install auto-py-to-exe_ from your terminal  

hope my comments will usefull for you.

**Reply by 1mkRE** [reply]
Hi i try to convert my customtkinter project to an exe file, but it still have an error that the "__init__.py , line 7 , in <module> import tkinter.filedialog as filedialog" cant find the tkinter.filedialog module.

At lease i get the Error : ModulNotFoundError: No module named 'tkinter.filedialog'

I do everything how the developer required, how to convert an py to exe but still have problems.
Please have anybody any idea?

tkinter is installed, it works in all my projects good but cant use customtkinter :-(

**Reply by Akascape** [reply]
@1mkRE If you are using pycharm and getting these module import errors, then have a look here: https://github.com/TomSchimansky/CustomTkinter/discussions/883#discussioncomment-4460667

Here is a full guide on packaging with pyinstaller: https://github.com/TomSchimansky/CustomTkinter/discussions/939

**Reply by 1mkRE** [reply]
@Akascape, i can't find any solution that descripes my problem. I packaging the application like you send me.
Below is the error which makes me trouble.

![Fehler1](https://user-images.githubusercontent.com/59584216/210150935-9b9ff25f-a559-465c-9701-a3ecdcb3387f.jpg)

**Reply by Akascape** [reply]
@1mkRE This one should help: 
https://github.com/TomSchimansky/CustomTkinter/discussions/896#discussioncomment-4449864

In short, your pyinstaller is not able to package modules from the python version because it is inside a virtual env. You have to change some settings in pycharm(if you are using that).

Otherwise you have to pack all the required module folders one by one.

**Reply by 1mkRE** [reply]
Thank you for your last reply, the problem above i think was solved.Thank you I just put step by step each required folder in the Add files.
But now i still fighting with an other problem with an other folder and modul see the picture attached.
Maybe you happen to have an idea how I can find out what the problem can be?

I try out find to run the app on different Windows systems with the same result. 
Furthermore I try out to find duplicated file names but not sure which folder to search exactly. 
At least I try out to put types.py into http folder where the cookies.py is filed, because the cookies.py have as reference import types.

Thanks.

![1](https://user-images.githubusercontent.com/59584216/210400891-67c83dec-450a-40ae-8320-324d49a67ab4.jpg)

**Reply by Akascape** [reply]
@1mkRE If you have a separate system (where python is not installed) then follow these steps:

1) Install a fresh python idle in windows (no virtual env, just python). 
2) Install the required modules like customtkinter and pyinstaller/Auto-py-to-exe.
3) Do the packaging for your app. This time it will collect modules automatically. 

Now you can use that system to pack your exe programs. This is the best way to make an exe. 
Otherwise using a virtual env with pyinstaller is a pain.

**Reply by 1mkRE** [reply]
@ Akascape, your solution works.
Today i tryed out on an other system without choosing an Python interpreter, just download my project from github, installed all required moduls and try out to convert the project according to the description and everything works.

I tryed out convert as Onefile, too and its working fine,too :-) (just importand to put images in same folder if required as a part of the project)

Thank you so much.

Best regards.
